stubs/parsimonious/parsimonious/expressions.pyi

```diff
@@ -0,0 +1,68 @@
+import typing
+from typing import Any, Callable, Pattern
+
+from parsimonious.exceptions import IncompleteParseError as IncompleteParseError, ParseError as ParseError
+from parsimonious.grammar import Grammar as Grammar
+from parsimonious.nodes import Node as Node, RegexNode as RegexNode
+from parsimonious.utils import StrAndRepr as StrAndRepr
+
+MARKER: Any
+
+def is_callable(value: Any) -> bool: ...
+
+_CALLABLE_RETURN_TYPE = int | tuple[int, list[Node]] | Node | None
+_CALLABLE_TYPE = (
+    Callable[[str, int], _CALLABLE_RETURN_TYPE]
+    | Callable[[str, int, dict[tuple[int, int], Node], ParseError, Grammar], _CALLABLE_RETURN_TYPE]
+    | Callable[[Any, str, int], _CALLABLE_RETURN_TYPE]
```

Comment:
At runtime `expression()` raises an error if it doesn't take either 2 or 5 arguments, so do we need the 3-arg and 6-arg variants?

Reply:
So these account for this case: https://github.com/erikrose/parsimonious/blob/3da7e804c07d4e495873be208701b5c955247c58/parsimonious/expressions.py#L72-L74

Essentially it's meant to cover when methods are supplied as the custom callable e.g. `callable=some_instance.some_method`. The code just assumes the first argument is `self` in that case and removes it and then checks the length is 2 or 5.

I put `Any` to represent this `self` param since it could be for any custom class. lmk if there's a better way to type it though 😄

Reply:
Actually apologies, just checked that code for the latest release of the package and these method arguments weren't available at that point so will remove them now: https://github.com/erikrose/parsimonious/blob/f4496cc0b6bae44c09eaaf4233e94a8d70e396a9/parsimonious/expressions.py#L60-L67